### DISCUSSION TITLE
Does solid-start-cloudflare-workers adapter unable to pnpm dev?

### DISCUSSION THREAD

**Discussion by dzolt** [solidjs/solid-start#788] [Q&A]
Hey! So I've installed solid-start-cloudflare-workers adapter to push my solid app to the cf workers and connect it to D1 db. 

I can run wrangler dev and it opens up almost fine (I have some issue with better-sqlite3 saying that it was compiled against differend node.js version and I also cannot resolve it) but it works.

However when I run command "pnpm dev" to start solid locally, miniflare throws an error seen in the screenshot below. Anyone has any idea why is that happening?
![image](https://user-images.githubusercontent.com/40149946/223128453-c0dc57ec-b878-4601-aa65-1a895cddd979.png)

**Comment by AlexErrant** [comment]
I use solid-start-cf, [and it works fine for me](https://github.com/AlexErrant/Pentive/blob/main/hub/package.json). Don't think I've run into your error.

```
"solid-start-cloudflare-workers": "^0.2.23"
"solid-start": "^0.2.23",
```

![image](https://user-images.githubusercontent.com/109672176/224546785-f96683a4-8d2e-4efd-9f9f-d9474e20e51c.png)

Sorry that I can't be more prescriptive with a solution.

I also had that better-sqlite3 node error. I resolved it by [pinning my node to 16.13.0](https://github.com/AlexErrant/Pentive/commit/ab06df3135dc4fe8c95d4473f8cc71937409fede). Note [this issue](https://github.com/WiseLibs/better-sqlite3/issues/797). If you're using [cr-sqlite](https://github.com/vlcn-io/cr-sqlite/), IIRC you need to either use the prebuilt or build on node19.

I don't recommend actually running my project - I desperately need to document how to Get Started. LMK if you wanna try; I'll need to do it sooner or later anyway x|

**Reply by dzolt** [reply]
Sorry for long reply. Unfortunately pinning my node to 16.13.0 isn't working and I get the same issue :/ And also the issue with better-sqlite has been resolved when I used 16.13.0. There seems to be some problem with solid-start miniflare but I cannot quite grasp why is it happening :/

**Reply by AlexErrant** [reply]
I [better documented how to get started with my repo](https://github.com/AlexErrant/Pentive/blob/main/design-decisions/run-locally.md), but I still don't recommend trying it (unless you really wanna play with HTTPS problems all day). Can you create a minimal example repo?
